main.tf

```diff
@@ -0,0 +1,135 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+
+# Configure the AWS Provider
+provider "aws" {
+  region = "us-east-1"
+}
+
+# Create a VPC
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+}
+
+locals {
+  public_cidr  = ["10.0.0.0/24", "10.0.1.0/24"]
+  private_cidr = ["10.0.2.0/24", "10.0.3.0/24"]
+}
+
+resource "aws_subnet" "public" {
+  count = 2
```

Comment:
This will only work in case we need two subnets per VPC.
However, very often we need 3 or 5 or even 7 subnets.
In order to optimize that and do fewer code changes, we can use the `length` function of terraform.

Instead of 
`count = 2` 
we can use 
`count = length(local.public_cidr)` 
or 
`count = length(local.private_cidr`

Please change your code so it uses the length function to calculate the proper count setting. (her in all other places that use count)

More info here: https://www.terraform.io/language/functions/length

Reply:
Updated counts to make dynamic using length function